src/agent/onefuzz-agent/src/managed/cmd.rs

```diff
@@ -13,7 +23,17 @@ pub async fn run(args: &clap::ArgMatches<'_>) -> Result<()> {
     let config = Config::from_file(config_path, setup_dir)?;
 
     init_telemetry(config.common());
-    let result = config.run().await;
+
+    let result = tokio::select! {
+        result = config.run() => result,
+
+        // Ignore this task if it returns due to a querying error.
+        Ok(oom) = out_of_memory(MIN_AVAILABLE_BYTES) => {
+            // Convert the OOM notification to an error, so we can log it below.
+            let err = format_err!("out of memory: {} bytes available, {} required", oom.available_bytes, oom.min_bytes);
+            Err(err)
```

Comment:
I think the decision to stop the task when OOM should be configurable and enabled by default.

Reply:
Added to the common config, on by default: 46a3d7a899ac90049fab00a3f04d1f3e71e4c222

Can be disabled by [setting to 0](https://github.com/microsoft/onefuzz/pull/1657/commits/46a3d7a899ac90049fab00a3f04d1f3e71e4c222#diff-083902b0f7ce9dceb9ecfb4eb6dda1de4a3595bd2161c306505e925880271b1cR55-R57).

Reply:
Added: https://github.com/microsoft/onefuzz/pull/1657/commits/0befdc5f5608141a0e8fc86122dc0b37e2c432b0